### PR TITLE
Update README.md to reflect v0.4.0

### PR DESCRIPTION
**Pull request by burningthumb**
I updated the read me to reflect, better, the current version from the maven repository.  It may not be perfect but I think it's at least better than what was there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,25 +20,113 @@ repositories {
 }
 
 dependencies {
-    implementation "com.github.nextcloud:Android-SingleSignOn:0.3.0"
+	implementation 'com.google.code.gson:gson:2.8.5'
```
Comment (tobiasKaminsky) [inline README.md]: If this is needed, it should be a dep from SSO

**Comment by tobiasKaminsky** [comment]
Thank you very much for your effort!

@David-Development can you please have a look as you are also using it as "client" and know the code best :-)

File: README.md (hunk 1 above)
Comment (burningthumb) [reply README.md]: The only thing I know is that the read me has this line of code in onActivityResult:

`NextcloudAPI nextcloudAPI = new NextcloudAPI(context, ssoAccount, new GsonBuilder().create(), callback);`

And if you don't include the implementation 'com.google.code.gson:gson:2.8.5' dependency it does not build.

Comment (David-Development) [reply README.md]: @tobiasKaminsky indeed.. would be better to include it as a dependency of the sso library as it is mandatory.

Comment (David-Development) [reply README.md]: Just opened #67 as a TODO to work on this.

**Comment by David-Development** [comment]
Looks good to me! @burningthumb Thanks for contributing and improving the docs! Hopefully you like this library! If you have any other suggestions on how to improve this library, let us know! :)

**Pull request merged**